### ISSUE TITLE
Retrieving profile data from LoginAccountsController::EVENT_BEFORE_REGISTER hook

### ISSUE THREAD

**Issue by adambcts** [dukt/social#23]
When using the LoginAccountsController::EVENT_BEFORE_REGISTER hook, the returned event object is a base Event and does not provide access to the profile data.

The solution would be to create another event as follows:

```
<?php
/**
 * @link      https://dukt.net/social/
 * @copyright Copyright (c) 2018, Dukt
 * @license   https://github.com/dukt/social/blob/v2/LICENSE.md
 */

namespace dukt\social\events;

use yii\base\Event;

/**
 * RegisterLoginProviderTypesEvent class.
 *
 * @author Dukt <support@dukt.net>
 * @since  2.0
 */
class LoginAccountsEvent extends Event
{
    // Properties
    // =========================================================================

    /**
     * @var The profile object.
     */
    public $profile;
}
```

And update the line number 488 in LoginAccountsController.php to use the new event and update 489 to return the key profile as follows:

```
            $this->trigger(self::EVENT_BEFORE_REGISTER, new LoginAccountsEvent([
                'profile' => &$profile,
```

For usage as follows:

```
Event::on(LoginAccountsController::class,
                LoginAccountsController::EVENT_BEFORE_REGISTER,
                function($event){
                    $email = $event->profile->getEmail();
                }
            );
```
Great plugin, forgive me if I have misunderstood something.

**Comment by adampatpattison** [comment]
[Pull Request 25](https://github.com/dukt/social/pull/25) solves this exact issue.

**Comment by benjamindavid** [comment]
Fixed by the now-merged #25. Thank you guys!
